src/components/app-configure/access/EditDelegate.vue

```diff
@@ -0,0 +1,152 @@
+<script lang="ts" setup>
+import { onMounted, ref } from 'vue'
+
+import VButton from '@/components/lib/VButton/VButton.vue'
+import VCard from '@/components/lib/VCard/VCard.vue'
+import VDropdown from '@/components/lib/VDropdown/VDropdown.vue'
+import VOverlay from '@/components/lib/VOverlay/VOverlay.vue'
+import VSeperator from '@/components/lib/VSeperator/VSeperator.vue'
+import VStack from '@/components/lib/VStack/VStack.vue'
+import VTextField from '@/components/lib/VTextField/VTextField.vue'
+import { useToast } from '@/components/lib/VToast'
+import { editDelegate } from '@/services/gateway.service'
+import { useAppId } from '@/use/getAppId'
+
+const emit = defineEmits(['close', 'generateKey', 'edited'])
+
+const props = defineProps({
+  delegateInfo: {
+    type: Object,
+    required: true,
+  },
+  delegateKeys: {
+    type: Array,
+    required: true,
+  },
+})
```

Comment:
Prefer using TypeScript for defining the prop types.

Reply:
https://github.com/arcana-network/developer-dashboard/pull/156/commits/764ff0c48c67ed42721b0494b417f70842081ebb